pyigm/cgm/cgm.py

```diff
@@ -323,10 +330,26 @@ def stack_plot(self,to_plot,pvlim=None,maxtrans=3,return_fig=True,**kwargs):
                 else: # Pick components in system closest to z_cgm
                     thesecomps = pu.get_components(self,tp)
                     if len(thesecomps)==0:
-                        continue
-                    # Find component with redshift closest to systemic
-                    compvels = np.array([np.median(tc.vlim.value) for tc in thesecomps])
-                    comp = thesecomps[np.argmin(np.abs(compvels))]
+                        if add_missing_lines is True:
+                            if isinstance(tp,str):
```

Comment:
I'm not sure str is 2.7 compatible, but
that's why we are on the way to 3.X

Reply:
It actually works under 2.7 on my desktop machine!